server/schedulers/hot_region.go

```diff
@@ -672,118 +696,118 @@ func (bs *balanceSolver) buildOperators() []*operator.Operator {
+			return true
 		}
+		return false
+	})
+
+	if len(stores) > 0 && loadDetail[stores[0]].LoadPred.Current.Count > 1 {
+		return stores[0]
 	}
-	return
+	return 0
 }
 
-// selectDstStoreByHot selects a target store to hold the region of the source region.
-// We choose a target store based on the hot region number and flow bytes of this store.
-func (h *hotScheduler) selectDstStoreByHot(candidates map[uint64]struct{}, regionBytesRate float64, srcStoreID uint64, storesStat statistics.StoreHotPeersStat) (dstStoreID uint64) {
-	srcBytesRate := storesStat[srcStoreID].FutureBytesRate
-	srcCount := len(storesStat[srcStoreID].Stats)
-	var (
-		minBytesRate = srcBytesRate*hotRegionScheduleFactor - regionBytesRate
-		minCount     = math.MaxInt32
-	)
-	for storeID := range candidates {
-		if s, ok := storesStat[storeID]; ok {
-			dstCount, dstBytesRate := len(s.Stats), math.Max(s.StoreBytesRate, s.FutureBytesRate)
-			// pick by count
-			if srcCount < dstCount+2 { // ensure srcCount >= dstCount after the operation.
-				continue
-			}
-			if minCount > dstCount || (minCount == dstCount && minBytesRate > dstBytesRate) {
-				dstStoreID = storeID
-				minBytesRate = dstBytesRate
-				minCount = dstCount
-			}
-		} else {
-			return storeID
+// Prefer store with larger `byteRate`.
+func selectSrcStoreByByteRate(loadDetail map[uint64]*storeLoadDetail) uint64 {
+	stores := sortStores(loadDetail, func(lp1, lp2 *storeLoadPred) bool {
+		ld1, ld2 := lp1.min(), lp2.min()
+		if ld1.ByteRate > ld2.ByteRate ||
+			(ld1.ByteRate == ld2.ByteRate && ld1.Count > ld2.Count) {
+			return true
+		}
+		return false
+	})
+
+	for _, id := range stores {
+		if loadDetail[id].LoadPred.Current.Count > 1 {
+			return id
 		}
 	}
-	return dstStoreID
+	return 0
 }
 
-func (h *hotScheduler) selectSrcStoreByScore(stats statistics.StoreHotPeersStat, balanceType rwType) uint64 {
-	var storesScore *ScoreInfos
-	if balanceType == read {
-		storesScore = h.readScores
-	} else {
-		storesScore = h.writeScores
-	}
-	maxScore := storesScore.Max()                 // with sort
-	for i := storesScore.Len() - 1; i >= 0; i-- { // select from large to small
-		scoreInfo := storesScore.scoreInfos[i]
-		if scoreInfo.GetScore() >= maxScore*hotRegionScheduleFactor {
-			storeID := scoreInfo.GetStoreID()
-			if stat, ok := stats[storeID]; ok && len(stat.Stats) > 1 {
-				return storeID
-			}
+// Prefer store with smaller `count`.
+func selectDstStoreByCount(candidates map[uint64]*storeLoadDetail, regionBytesRate float64, srcLoadDetail *storeLoadDetail) uint64 {
+	stores := sortStores(candidates, func(lp1, lp2 *storeLoadPred) bool {
+		ld1, ld2 := lp1.max(), lp2.max()
+		if ld1.Count < ld2.Count ||
+			(ld1.Count == ld2.Count && ld1.ByteRate < ld2.ByteRate) {
+			return true
+		}
+		return false
+	})
+
+	srcLoad := srcLoadDetail.LoadPred.min()
+	for _, id := range stores {
+		dstLoad := candidates[id].LoadPred.max()
+		if srcLoad.Count-1 >= dstLoad.Count+1 &&
```

Comment:
before is `> 2` ,  and changed to `>=2`?

Reply:
Yes.